### ISSUE TITLE
HTTP 429 Too Many Requests.

### ISSUE THREAD

**Issue by aminul-xs** [ai-genie/chatgpt-vscode#51]
### Verify it's not a duplicate bug report

- [X] I searched for similar bugs reported on GitHub and nobody has reported a similar issue. <a href="https://github.com/ai-genie/vscode-chatgpt/issues?q=" target="_blank">Search all issuees</a>.
- [X] I read <a href="https://github.com/ai-genie/vscode-chatgpt#troubleshooting" target="_blank">Troubleshooting guide on Readme</a> and there is not a guide on how to solve my problem.

### Describe the Bug

HTTP 429 Too Many Requests. Too many requests try again later. You can clear your API Key from VS Code's [secrets storage](https://code.visualstudio.com/api/references/vscode-api#SecretStorage) with Genie: Clear API Key command. Potential reasons:

You exceeded your current quota, please check your plan and billing details
You are sending requests too quickly
The engine is currently overloaded, please try again later.
See https://platform.openai.com/docs/guides/error-codes for more details.
{
    "error": {
        "message": "You exceeded your current quota, please check your plan and billing details.",
        "type": "insufficient_quota",
        "param": null,
        "code": null
    }
}

### Please tell us if you have customized any of the extension settings or whether you are using the defaults.

default

**Comment by Karest2512** [comment]
![image](https://user-images.githubusercontent.com/78431084/229441135-850d6a1e-5202-4668-8d93-f5fb8be5eafe.png)
I am also facing this. I think getting this error may be because the free account has run out of money for the trial. Excuse me because my English is not good.

**Comment by cbachen1997** [comment]
I am getting this error too. Idk whether the reason is i am using a VPN. 
But the chatGPT on web still works normally, it seems to be a problem of API.

**Comment by cazzoo** [comment]
Facing the exact same here. I couldn't get it working yet

**Comment by sp1ke1980** [comment]
I have the same issue in vscode.

**Comment by genieai-info** [comment]
Hi, As the error states you have `insufficient_quota` error thrown from OpenAI side, it is out of extension's control. However, here are some steps that may help you understand your issue:
1. Make sure your account hasn't expired from here: https://platform.openai.com/account/usage
2. Make sure you have enough credit to use OpenAI APIs from here: https://platform.openai.com/account/usage
3. Try generating a new API Key and change the API Key used in the extension from here: https://platform.openai.com/account/api-keys. To change your API Key, Click on `Change API Key` on the home page of the sidebar Genie: Conversation view.

If you have an active account with credit it will show on the /usage page like this:

![image](https://user-images.githubusercontent.com/128560534/229565363-2b7e16fb-8544-4f64-9e03-3b4062b3ed59.png)

**Comment by viking1304** [comment]
Guys, please try to search for a similar issue before posting.

Most developers (including me) would just close this issue with a remark that this is a duplicate since this is exactly the same issue as https://github.com/ai-genie/chatgpt-vscode/issues/25

@genieai-info when someone is trying to post a new issue, they currently have those two checkboxes

```
[ ] I searched for similar bugs reported on GitHub and nobody has reported a similar issue. [Search all issuees](https://github.com/ai-genie/vscode-chatgpt/issues?q=). *

[ ] I read [Troubleshooting guide on Readme](https://github.com/ai-genie/vscode-chatgpt#troubleshooting) and there is not a guide on how to solve my problem. *
```

Adding a new section with some additional checkboxes might be a good idea.


### Additional checks in case of 429 Too Many Requests error 
Confirm that you did check and understood all of the following

[ ] I have read the [ChatGPT IDE FAQ](https://help.openai.com/en/articles/7039783-chatgpt-api-faq), and I am aware that OpenAI IDE is not part of the ChatGPT Plus subscription and that I need to pay for it separately

[ ] I have checked that my [account](https://platform.openai.com/account/usage) is not expired

[ ] I have checked that I have enough credit on my [account](https://platform.openai.com/account/usage) 

[ ] I have tried to use cURL, and I am not getting the 429 error with cURL

**Comment by sp1ke1980** [comment]
Good point, but I've tested with different api keys and problem just only with your extension. When I'm doing request with other application, the problem is not appearing

**Comment by antisect** [comment]
was getting this error, signed up for a brans new chatGPT account and new api key and im still getting this error

**Comment by Salmagic** [comment]
i also made a brand new account and was still getting the error, it was only when i set up a new phone number then it stared working again, but it says i got a 5USD credit which will expire fast, so i guess the problem was not having enough credit.. sadly

**Comment by yxw007** [comment]
> 

Clean up the key first, and then add a new key. try again

**Comment by xiaoouwang** [comment]
clean and add a new key, doesn't work. I have chatgpt plus api.

**Comment by fblanc92** [comment]
Same here. Cleaned, created a new one, still doesn't work

**Comment by skossan** [comment]
I also have the same issue with this extension. Cleaned API key and added a new one, still the same issue.

**Comment by ianccy** [comment]
Got the same problem today.

i have tried to reset the new api key but it's still not work.

**Comment by uekky** [comment]
I have remedied this problem by adding a payment method.

**Comment by ianccy** [comment]
i aslo found the way to solve the problem. The way is register the new account...

**Comment by tn00251521** [comment]
Same here. Cleaned, created a new one, still doesn't work

**Comment by piavgh** [comment]
same here, still working fine on chat.openai.com, but using Genie, it does not work

**Comment by viking1304** [comment]
Guys, I am sorry to be "that" guy, but with all due respect, is it hard to read what other people write and check a few simple things?

You have to understand that you need a free trial of OpenAI API or pay for it. It is not part of ChatGPT Plus, and if chat.openai.com works for you, that doesn't mean you have the right to use API and that it will work.

Even if you didn't spend any of your credit
<img width="126" alt="Screenshot 2023-04-07 at 09 59 51" src="https://user-images.githubusercontent.com/28989870/230568280-97ea276b-6110-409c-b973-aee6393e50c6.png">

you can't use the API if your trial is expired
<img width="209" alt="Screenshot 2023-04-07 at 09 59 36" src="https://user-images.githubusercontent.com/28989870/230568267-ed14e231-7e4c-41d4-ba26-e20c628fb12d.png">

and you will get 429 responses every single time, even if you create 100 new keys.

I was also confused at first and complained that the previous plugin didn't work for me until I finally realized I was wrong from the beginning.

The only way to help developers is to resolve the bug, if it exists, is to understand this message, check everything on your side, test your key with cULR and post the response if it is anything other than 429.

Please do all of those:

1. read the [ChatGPT API FAQ](https://help.openai.com/en/articles/7039783-chatgpt-api-faq)
2. check your [Usage limits](https://platform.openai.com/account/billing/limits). "You must be on a paid plan to set up usage limits" means that you are using a free API trial, even if you are ChatGPT Plus subscriber
3. if you didn't pay for API, check your [account](https://platform.openai.com/account/usage) and see if your trial is expired
4. on the same page, check that you have  enough credit if the trial is not expired
5. if everything is fine there, check with cURL and post the result

**Comment by viking1304** [comment]
> clean and add a new key, doesn't work. I have chatgpt plus api.

I am sorry, but I have to tell you the sad truth. API usage is not included in your ChatGPT Plus subscription. You need to pay for API or to use the free trial if you want to use API. Read the [ChatGPT API FAQ](https://help.openai.com/en/articles/7039783-chatgpt-api-faq)

**Comment by xiaoouwang** [comment]
> 1. https://platform.openai.com/account/usage



> Guys, I am sorry to be "that" guy, but with all due respect, is it hard to read what other people write and check a few simple things?
> 
> You have to understand that you need a free trial of OpenAI API or pay for it. It is not part of ChatGPT Plus, and if chat.openai.com works for you, that doesn't mean you have the right to use API and that it will work.
> 
> Even if you didn't spend any of your credit <img alt="Screenshot 2023-04-07 at 09 59 51" width="126" src="https://user-images.githubusercontent.com/28989870/230568280-97ea276b-6110-409c-b973-aee6393e50c6.png">
> 
> you can't use the API if your trial is expired <img alt="Screenshot 2023-04-07 at 09 59 36" width="209" src="https://user-images.githubusercontent.com/28989870/230568267-ed14e231-7e4c-41d4-ba26-e20c628fb12d.png">
> 
> and you will get 429 responses every single time, even if you create 100 new keys.
> 
> I was also confused at first and complained that the previous plugin didn't work for me until I finally realized I was wrong from the beginning.
> 
> The only way to help developers is to resolve the bug, if it exists, is to understand this message, check everything on your side, test your key with cULR and post the response if it is anything other than 429.
> 
> Please do all of those:
> 
> 1. read the [ChatGPT API FAQ](https://help.openai.com/en/articles/7039783-chatgpt-api-faq)
> 2. check your [Usage limits](https://platform.openai.com/account/billing/limits). "You must be on a paid plan to set up usage limits" means that you are using a free API trial, even if you are ChatGPT Plus subscriber
> 3. if you didn't pay for API, check your [account](https://platform.openai.com/account/usage) and see if your trial is expired
> 4. on the same page, check that you have  enough credit if the trial is not expired
> 5. if everything is fine there, check with cURL and post the result

Yeah, now I see why. It's a twisted and tricky move from openai to make these two billed separately. That's why a lot of people got confused I guess even with the instructions above.

So yes,

> Is the ChatGPT API included in the ChatGPT Plus subscription?: No, the ChatGPT API and ChatGPT Plus subscription are billed separately.

**Comment by viking1304** [comment]
> Yeah, now I see why. It's a twisted and tricky move from openai to make these two billed separately. That's why a lot of people got confused I guess even with the instructions above.

I have a feeling that a lot of Plus subscribers would never pay for it if they knew that they cannot use API. Also, in most cases, it is also much cheaper to pay for API than to pay for a Plus subscription.

**Comment by xiaoouwang** [comment]
Yes I just calceled my subscription. Will test how much I consumed with the
api. Thanks to this extension 🤣

**Comment by oywino** [comment]
How come my free API account expired on April 1st when it was created today (April 14) ?!?
Did it expire backwards in time?
Is this a negative free time? 😶

![image](https://user-images.githubusercontent.com/18447119/232047082-d04ef58a-ec75-4af9-9757-c63a4e6ef660.png)

**Comment by kennyan11b** [comment]
> How come my free API account expired on April 1st when it was created today (April 14) ?!? Did it expire backwards in time? Is this a negative free time? 😶
> 
> ![image](https://user-images.githubusercontent.com/18447119/232047082-d04ef58a-ec75-4af9-9757-c63a4e6ef660.png)

Same. I just installed Genie yesterday on VSCode and it said I have expired credits which expired on the 1st.

**Comment by viking1304** [comment]
@oywino and @kennyan11b - if you applied for a DALL·E access before, your API trial started when you got confirmation mail that you can try DALL·E

Check your emails and see if you have a mail titled "Your invite to create with DALL·E is here." If you have it check the date.

btw. new accounts will get only $5.00 credits instead of $18.00. If you see $18.00, you can be 99.99999999% sure that your trial is already expired.

**Comment by oywino** [comment]
Hi, I decided to sign up, and mine now looks like this:

![image](https://user-images.githubusercontent.com/18447119/232467297-dbfae733-55c9-49b4-9222-9939a758b924.png)

**Comment by RonaldGuilhermePDS** [comment]
Hey, Is there a way to use this extension without needing a login?

**Comment by faqcodes** [comment]
<img width="793" alt="image" src="https://user-images.githubusercontent.com/124191313/233122561-1247028b-8e29-4353-9a95-a8ffdda666db.png">

**Comment by VyacheslavKolodrubskiy** [comment]
> Hey, Is there a way to use this extension without needing a login?

You can use ChatGPT - EasyCode, no account or API key required

**Comment by programmerrmasud** [comment]
Create a New GPT account and log in by using this one. It will be solved 100% .

**Comment by maheshpansare** [comment]
I'm facing the same issue with HTTP 429 even though I have a credit of $2500 and have paid a subscription.

**Comment by matthewbreisch** [comment]
What a scam. This whole thing left a bad taste in my mouth.
